dashboard/app/controllers/aichat_controller.rb

```diff
@@ -103,20 +103,55 @@ def log_chat_event
         script_id: context[:scriptId],
         project_id: project_id,
         request_id: event[:requestId], # Only present if ChatEvent is a ChatMessage, otherwise nil
-        aichat_event: event.to_json
+        aichat_event: event
       )
     rescue StandardError => exception
       return render status: :bad_request, json: {error: exception.message}
     end
 
     response_body = {
-      chat_event_id: logged_event.id,
-      chat_event: logged_event.aichat_event
+      id: logged_event.id,
+      **logged_event.aichat_event
     }
 
     render(status: :ok, json: response_body)
   end
 
+  # params are eventId: number, feedback?: 'clean_disagree' | 'profanity_agree' | 'profanity_disagree'
+  # POST /aichat/submit_teacher_feedback
+  # Update a given chat message with teacher feedback. If feedback is nil, remove any existing feedback.
+  # Also has the side effect of fixing up any chat events that were stored as strings.
```

Comment:
Ticket to clean this up ~3 months from now since data is deleted after 90 days (lines 143, 173). https://codedotorg.atlassian.net/browse/LABS-1324